src/thing_shots.c

```diff
@@ -472,9 +476,42 @@ TbBool shot_hit_wall_at(struct Thing *shotng, struct Coord3d *pos)
                 apply_damage_to_thing(doortng, i, shotst->damage_type, -1);
             } else
             {
-                efftng = create_shot_hit_effect(&shotng->mappos, shotng->owner, shotst->hit_generic.effect_model, shotst->hit_generic.sndsample_idx, shotst->hit_generic.sndsample_range);
-                if (!shotst->hit_generic.withstand)
+                long smpl_idx;
+                long range;
+                long eff_kind;
+                struct Coord3d pos;
+                long eff_owner;
+                if (digging)
                 {
+                    struct SlabMap* slb = get_slabmap_for_subtile(stl_num_decode_x(hit_stl_num), stl_num_decode_y(hit_stl_num));
+                    if ( (old_health > slb->health) || (slb->kind == SlbT_GEMS) )
+                    {
+                        smpl_idx = 63;
```

Comment:
With these changes we'd have these values hardcoded at two spots. Ones is too often, twice is worse still.
I think we could add `struct ShotHitConfig hit_digging;` to `struct ShotConfigStats` in `config_magic.h`, add these values into magic.cfg on SHOT_DIG, and use them not just for possession but for `instf_dig` as well.